### ISSUE TITLE
semi_join() erros

### ISSUE THREAD

**Issue by antonio-yu** [machow/siuba#351]
```
from siuba import * 
from siuba.data import mtcars 

mtcars >> semi_join(_,mtcars)
 # Check for duplicates
KeyError: {'gear', 'carb', 'cyl', 'mpg', 'hp', 'qsec', 'wt', 'am', 'vs', 'disp', 'drat'}
```

**Comment by antonio-yu** [comment]
@machow  Hello, could you help seeing this issue?

**Comment by machow** [comment]
Hey--it looks like when no `on` columns are specified, semi_join tries to guess them. Unfortunately, it makes the guessed keys a set, which seems to not work in the merge.

Should be able to fix by specifying the columns to join on. I'm going to try and publish the stable branch as a release candidate today on pypi, and then can try a quick fix!

https://github.com/machow/siuba/blob/master/siuba/dply/verbs.py#L1093

**Comment by machow** [comment]
Ah, sorry--released 0.1.2 just now which should have the fix (note that I reverted the versioning back to 0.1.2, to make it easier for people to pip install by default, versus using alpha releases #353 )
